daphne/src/roles_test.rs

```diff
@@ -559,16 +690,157 @@ async fn poll_collect_job_test_results() {
     );
 }
 
+#[tokio::test]
+async fn http_post_collect_fail_invalid_batch_interval() {
+    let leader = MockAggregator::new();
+    let task_id = leader.nominal_task_id();
+    let task_config = leader.get_task_config_for(task_id).unwrap();
+    let now = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    // Collector: Create a CollectReq with a very large batch interval.
+    let collector_collect_req = CollectReq {
+        task_id: task_id.clone(),
+        batch_interval: Interval {
+            start: now - (now % task_config.min_batch_duration),
+            duration: task_config.min_batch_duration
+                * (leader.global_config.max_batch_duration + 1),
+        },
+        agg_param: Vec::default(),
+    };
+    let req = DapRequest {
+        media_type: Some(MEDIA_TYPE_COLLECT_REQ),
+        payload: collector_collect_req.get_encoded(),
+        url: task_config.helper_url.join("/collect").unwrap(),
+        sender_auth: Some(BearerToken::from(COLLECTOR_BEARER_TOKEN.to_string())),
+    };
+
+    // Leader: Handle the CollectReq received from Collector.
+    let err = leader.http_post_collect(&req).await.unwrap_err();
+
+    // Fails because the requested batch interval is too large.
+    assert_matches!(err, DapAbort::InvalidBatchInterval);
```

Comment:
The "invalidBatchInterval" has a very specific semantics in DAP-01, this is technically non-compliant. We should consider putting this in the spec, but in the mean time I think we need to handle this explicitly. Can we return `DapAbort::BadRequest` with a string that signifies why we're rejecting the report?

Reply:
I've used BadRequest and used the string to explicitly report about the reason.